### PR TITLE
Join room other federation via alias error

### LINKED ISSUE #2778: Join room by alias other federation error
Step to reproduce:

From a matrix.org account try to join this room #cloud-devroom_testing4:fosdem.org

=> android displays a "This room is not accessible at this time", and doesn't give you an option to try and join?

### PR DESCRIPTION
**Pull request by BillCarsonFr**
Fixes #2778
Peek room task was not catching enough failures (of child tasks call)

### CODE REVIEW

**Review by ganfra — APPROVED** [review]
LGTM, thanks

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/peeking/PeekRoomTask.kt [hunk 1]
```diff
@@ -65,23 +66,29 @@ internal class DefaultPeekRoomTask @Inject constructor(
         }
 
         // Is it a public room?
-        val publicRepoResult = when (getRoomDirectoryVisibilityTask.execute(GetRoomDirectoryVisibilityTask.Params(roomId))) {
-            RoomDirectoryVisibility.PRIVATE -> {
-                // We cannot resolve this room :/
-                null
-            }
-            RoomDirectoryVisibility.PUBLIC  -> {
+        val visibilityRes = tryOrNull("## PEEK: failed to get visibility") {
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/peeking/PeekRoomTask.kt:69]: I'm not a big fan of using `tryOrNull {}` around tasks which make network request. If there is no network, there is no need to go further.

Comment (ganfra) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/peeking/PeekRoomTask.kt:69]: We don't have good mechanism to be sure we can reach the server, so how do you want to handle this?

Comment (bmarty) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/peeking/PeekRoomTask.kt:69]: No network: special Exception. Getting 404 is something else for instance

**Pull request merged**